test/Grpc.Net.Client.Tests/Balancer/ResolverTests.cs

```diff
@@ -127,17 +131,23 @@ public async Task Resolver_MatchingPolicyInResult_LoadBalancerUnchanged()
             await Resolver_ServiceConfigInResult(resolvedServiceConfig: new ServiceConfig
             {
                 LoadBalancingConfigs = { new PickFirstConfig() }
-            });
+            }, resolvedServiceConfigStatus: null);
+        }
+
+        [Test]
+        public void ResolverResult_OkStatusWithNoResolver_Error()
+        {
+            Assert.Throws<ArgumentException>(() => ResolverResult.ForResult(new List<BalancerAddress> { new BalancerAddress("localhost", 80) }, null, Status.DefaultSuccess));
         }
 
-        private async Task Resolver_ServiceConfigInResult(ServiceConfig? resolvedServiceConfig)
+        private async Task Resolver_ServiceConfigInResult(ServiceConfig? resolvedServiceConfig, Status? resolvedServiceConfigStatus)
```

Comment:
```suggestion
        private async Task Resolver_ServiceConfigInResult(ServiceConfig? resolvedServiceConfig, Status? resolvedServiceConfigStatus = null)
```

Then you don't need to update most of the invocations...